File: identhub/src/main/java/de/solarisbank/identhub/domain/contract/AuthorizeContractSignUseCase.java
package de.solarisbank.identhub.domain.contract;

import de.solarisbank.sdk.domain.usecase.CompletableUseCase;
import io.reactivex.Completable;
import kotlin.Unit;

public class AuthorizeContractSignUseCase implements CompletableUseCase<Unit> {
    private final ContractSignRepository contractSignRepository;

    public AuthorizeContractSignUseCase(ContractSignRepository contractSignRepository) {
        this.contractSignRepository = contractSignRepository;
    }

    @Override
    public Completable execute(Unit param) {
        return contractSignRepository.getIdentification()
                .flatMap(identification -> contractSignRepository.authorize(identification.getId()))
                .ignoreElement();
    }
}

File: identhub/src/main/java/de/solarisbank/identhub/domain/contract/AuthorizeContractSignUseCase.kt
package de.solarisbank.identhub.domain.contract

import de.solarisbank.identhub.data.contract.ContractSignRepository
import de.solarisbank.sdk.domain.usecase.CompletableUseCase
import io.reactivex.Completable

class AuthorizeContractSignUseCase(
    private val contractSignRepository: ContractSignRepository
    ) : CompletableUseCase<Unit> {
    override fun execute(param: Unit): Completable {
        return contractSignRepository.getIdentification()
            .flatMap { (id) -> contractSignRepository.authorize(id) }
            .ignoreElement()
    }
}